crates/runtime/src/traphandlers.rs

```diff
@@ -170,19 +171,150 @@ cfg_if::cfg_if! {
                 }
             }
         }
-    } else if #[cfg(target_os = "windows")] {
-        use winapi::um::errhandlingapi::*;
-        use winapi::um::winnt::*;
-        use winapi::um::minwinbase::*;
-        use winapi::vc::excpt::*;
+    } else if #[cfg(target_os = "fuchsia")] {
+        use fuchsia_zircon_sys::{self as sys, ZX_HANDLE_INVALID, ZX_OK};
+        use std::mem::MaybeUninit;
+
+        extern "C" {
+            pub fn zx_thread_self() -> sys::zx_handle_t;
+            pub fn zx_process_self() -> sys::zx_handle_t;
+        }
 
         unsafe fn platform_init() {
-            // our trap handler needs to go first, so that we can recover from
-            // wasm faults and continue execution, so pass `1` as a true value
-            // here.
-            if AddVectoredExceptionHandler(1, Some(exception_handler)).is_null() {
-                panic!("failed to add exception handler: {}", io::Error::last_os_error());
+            // TODO Currently we spawn one exception handling thread per runtime thread
```

Comment:
FWIW I think this isn't quite right, this is actually only one exception handling thread for the first thread, but no other threads get exception handlers set.